### ISSUE TITLE
2.15.10 Release checklist

### ISSUE THREAD

**Issue by devincowan** [hydroshare/hydroshare#5612]
### Issues in Release
- [x] #5511 by @devincowan 
- [x] #5635 by @devincowan 
- [x] #5566 by @devincowan 
- [x] #5654 by @devincowan 
- [x] #5676 by @devincowan 
- [x] #5677 by @devincowan 
- [x] #5597 by @devincowan 

### Recipies
https://github.com/CUAHSI/platform-recipes/pull/57

### Beta Deployment
- [x] Diff RC to master to identify and update helm chart for the following files:
  * scripts/templates/docker-compose.template
  * hydroshare/settings.py
  * hydroshare/local_settings.py
- [x] Determine whether maintenance page will be needed (db migrations etc)
- [x] Add any management commands necessary for the deploy to the notes section
- [x] Deployed to Beta
- [x] Review the search and discovery pages
- [x] Create a new user and update profile
- [ ] Create iROD account, test connection and delete iROD account
- [x] Create a new resource, check sharing/permission settings, delete new resource
- [x] Developers test around issues
- [x] [Hsclient tests pass when targeting beta](https://github.com/hydroshare/hsclient/actions/runs/11880975428)
- [x] QA testing around issues
- [ ] Stakeholders approval

### Production Deployment
- [x] Deployed to Production
  * Make manual changes to files identified in Beta Deployment
- [x] Review the search and discovery pages
- [ ] Create a new user and update profile
- [ ] Create iROD account, test connection and delete iROD account
- [x] Create a new resource, check sharing/permission settings, delete new resource
- [x] Developers test around issue

### Notes relevant to deployment
1. [Updates to settings](https://github.com/hydroshare/hydroshare/compare/2.15.9.1...rc-2.15.10#diff-2521ce60a97c1524722701093e6ea98b2ae1f1b819aaa8bf38068d2276e61de4)
2. Requires the following https://github.com/CUAHSI/platform-recipes/pull/57

### Diff
https://github.com/hydroshare/hydroshare/compare/2.15.9.1...rc-2.15.10

**Comment by dtarb** [comment]
For large file upload consider changing the hover over "remove file" to "Cancel" as the functionality seems to be to cancel the upload.
![image](https://github.com/user-attachments/assets/0d3addfd-4fc8-4817-9c4b-12d52547478b)

**Comment by dtarb** [comment]
Also on resuming an upload session, is there a way to offer it to a user sooner than when they click into edit and upload files.  Right now if I close the browser then log back in I have to go to the resource, go to edit mode, and click on add files before I see that there is a suspended upload.  I think that the option to resume a suspended upload should appear for a user as soon as they log in, or else they may not be aware that they have suspended uploads and suspended uploads may hang around indefinitely (or until they are cleaned out)

**Comment by devincowan** [comment]
@dtarb thanks!
RE [first comment](https://github.com/hydroshare/hydroshare/issues/5612#issuecomment-2498486534) comment:
How about instead of "Remove file" we instead say "Remove file from upload queue" ???
![image](https://github.com/user-attachments/assets/d163f7f2-2e64-49f4-84f9-346f5fe78078)

Re [second comment](https://github.com/hydroshare/hydroshare/issues/5612#issuecomment-2498497717)
Great suggestion to notify for recovered uploads! I added a resource specific notification:
![image](https://github.com/user-attachments/assets/7dd32475-7648-4ba1-8ec5-a757944f665b)
this alerts for upload attempts that were conducted specifically to the resource being viewed.
This notification will show immediately on viewing the resource after a failed upload attempt.

Notification will be shown at the resource level -- so if I login and go to the homepage after a failed upload attempt, I will not see the warning. Notifying on non-resource pages would require integrating uppy at a lower level in our templates, which I would not advise we do

**Comment by dtarb** [comment]
@devincowan I guess I am thinking of it differently than you.  I suggested "Cancel" for consistency with the cancel button at the bottom
![image](https://github.com/user-attachments/assets/d4a4e091-9dca-44bd-9a8f-b4095155ffb2)
I think"Remove file from upload queue" is too wordy and cryptic and "Cancel" is pretty clear.

Also, I think that if an upload has been interrupted and is not complete, I'd like to be notified as soon as possible, which is whenever I next log in.  I do not want to have to go back to that resource, which means I have to know that an upload for that resource was interrupted.  The consideration of where uppy is integrated is a software consideration, not a user experience consideration and a clean and simple user experience should be king. Also we already have a notifications bell top right.  I think that stylistically it may be better to use that for notifications of incomplete uploads rather than a red box that just appears.  But I would be interested in what good GUI UX practise says about this.  

I feel that overall we have a bit too many messages that appear at random positions on pages.  One I just noticed is 
![image](https://github.com/user-attachments/assets/078c71a5-a64c-4c42-b7f0-b90ed6b04dad)
Which really does not say much of use.  Perhaps this motivates an overall revamp of how clean and clear the UI is to address these.

**Comment by devincowan** [comment]
Thanks @dtarb I understand your points.

### Regarding the "cancel" vs "remove" buttons: 
The cancel button on each file preview is distinct from the cancel button at the bottom of the modal window. Uppy works as a two stage process:
  1. First files are selected and added the the staging area (what I have been calling the "queue").
  2. Upload is initiated and the staged files are uploaded in batches

The "X" button on each file preview in the modal (the one to which you propose adding a "cancel" tooltip) has the functionality of removing a staged file from the upload queue **BEFORE** the upload is initiated. The concept of "canceling" this single file doesn't really make sense -- it's upload has not actually been initiated yet.

The "X" button at the bottom of the modal is placed next to a "pause" button. These are the actions that can be taken **AFTER** upload is initiated -- ex: after I click green "upload 1 file"
![image](https://github.com/user-attachments/assets/b299cb70-360d-4445-9321-6046052b12d9)

The distinction between these "X" buttons is much more obvious when you have staged multiple files for an upload session:
![image](https://github.com/user-attachments/assets/ee905d31-b7c7-4ff1-b40f-9d71eeac0332)


I do agree that "remove file from upload queue" is too verbose. My preference is the Uppy default "Remove File" but I would also support "Unstage" or "De-queue". Uppy uses "Remove File" as in "Remove this file from the staging area"

**Comment by devincowan** [comment]
@dtarb thanks also for the feedback about notifications for interrupted uploads!

### Re interrupted upload notifications

When you log back in (after a catastrophic browser crash etc), if you go to the resource landing page on which you had initiated uploads:
![image](https://github.com/user-attachments/assets/f2a0325e-bb0c-4b17-87cf-a9e63c4999a5)

If you log back in and for some reason go to a different resource landing page than the one that you had initiated uploads on:
![image](https://github.com/user-attachments/assets/86079370-4182-4243-a2cb-add438d4cbe4)

I also did the work to integrate uppy at the root level of our templates so that the above notification can be shown on a non-resource page (for example on the home page) after a browser failure.

These changes have been deployed to beta

**Comment by dtarb** [comment]
Thanks for explaining the various cancel/close buttons. I see the distinction between "Remove file" on a file and Cancel and Close Modal.  

If it is possible to disable "Close Modal" while a file upload is in progress do so, because it results in an interrupted download.  It would be better to have a user deliberately click Cancel if they want to exit.  In the below
![image](https://github.com/user-attachments/assets/703d341b-0687-4be4-a6af-9430e9cdc1fa)
Cancel top left, Cancel bottom Right and Remove File all have the same behavior resulting in returning to the file selection screen, but Close Modal top right results in an interrupted upload.

Also at the first modal screen below, there is redundancy between browse files and import from "My Device"
![image](https://github.com/user-attachments/assets/5e4e10d6-e0b5-4fa5-b36d-5f8f156bc6dd)
It would be better if this could be avoided.

Also try avoid contradictions. E.g. after restoring an upload I got to the below which says "Upload Complete" but "You can now resume the upload" both on the same window.  The session restored you can resume the upload needs to go away when the upload completes.
![image](https://github.com/user-attachments/assets/3245d4bd-fa74-4dd2-9a3f-088ab8f0fb24)

Also, in the below it says "Upload 2 files at the bottom" but "1 file selected" at the top.  Since one file has been successfully uploaded already it should say "Upload 1 file" at the bottom.

![image](https://github.com/user-attachments/assets/db500c40-f215-4469-82f6-625598cbc036)

I realize that some of these may be unchangeable parts of uppy that we just have to live with, but felt it worth pointing them out in case it is possible to fix.
